app/src/main/java/com/example/erabook/util/Extensions.kt

```diff
@@ -31,4 +36,9 @@ fun ImageView.loadImageFromUrl(imageLink: String?) {
 
 fun Context.showToast(@StringRes messageResId: Int, duration: Int = Toast.LENGTH_SHORT) {
     Toast.makeText(this, messageResId, duration).show()
+}
+fun openLinkBrowser(url: String?, context: Context) {
+    val intent =
+        Intent(Intent.ACTION_VIEW, Uri.parse("https://www.barnesandnoble.com/s/$url"))
```

Comment:
You are not going to open always same URL, do you? So you need to pass the URL as the parameter to have it dynamic for other URLs.

Reply:
in this case I am going to open the same url but with different parameters at the end of it like the book name, because the api doesn't provide a link where you can buy the book directly. The only thing that will change in this url is the search parameter at the end - the book name

Reply:
Makes sense!